### PR TITLE
Documentation for ONNX export of Megatron Models

### PR DESCRIPTION
**Pull request by asfiyab-nvidia**
# What does this PR do ?

Adds documentation for ONNX export of Megatron Models

### CODE REVIEW

**Comment by ericharper** [comment]
Could you add this to index.rst so it shows up in the navigation pane?

**Comment by asfiyab-nvidia** [comment]
@ericharper would the `docs/source/nlp/nemo_megatron`  be a better location for this file? I can then include it in the intro.rst under `nemo_megatron`

**Comment by ericharper** [comment]
I think it's better at a higher level

**Comment by ericharper** [comment]
I'm thinking below NeMo Megatron API: 
image

**Comment by asfiyab-nvidia** [comment]
Added the change. Is that what you had in mind?

**Comment by ericharper** [comment]
> Added the change. Is that what you had in mind?

That is a good change.

But also add it to the table of contents: `docs/source/index.rst`

```
.. toctree::
   :maxdepth: 3
   :caption: Natural Language Processing
   :name: Natural Language Processing

   nlp/nemo_megatron/intro
   nlp/machine_translation/machine_translation
   nlp/text_normalization/intro
   nlp/api
   nlp/models
   ```

**Comment by asfiyab-nvidia** [comment]
Ah I see, just included it in index.rst as well

File: docs/source/nlp/megatron_onnx_export.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+.. code-block:: bash 
+
+    https://github.com/NVIDIA/TransformerEngine.git && cd TransformerEngine
+    git submodule update --init --recursive
+    python3 -m pip install -e .
+
+    cd ..
+    git clone https://github.com/NVIDIA/NeMo.git && cd NeMo/
+    python3 -m pip install -e .
```
Comment (ericharper) [inline docs/source/nlp/megatron_onnx_export.rst:26]: We have installation instructions already. Maybe you can just mention a tagged minimum version of TE and NeMo that is required?

Comment (asfiyab-nvidia) [reply docs/source/nlp/megatron_onnx_export.rst]: Done

File: examples/nlp/language_modeling/megatron_ckpt_to_nemo.py [hunk 2]
```diff
@@ -143,7 +143,15 @@ def convert(local_rank, rank, world_size, args):
...
     if args.model_type == 'gpt':
+        # model = MegatronGPTModel.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
         model = MegatronGPTModel.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
+
+        # WAR to enable exporting FP8 scales from modules
+        # TODO: Need to find a cleaner way
+        for name, module in model.named_modules():
+            if hasattr(module, 'fp8'):
+                setattr(module, 'fp8', True)
+
```
Comment (ericharper) [inline examples/nlp/language_modeling/megatron_ckpt_to_nemo.py:154]: Let's leave this PR for docs only

Comment (asfiyab-nvidia) [reply examples/nlp/language_modeling/megatron_ckpt_to_nemo.py:154]: Oops that change was included accidentally. Reverting

File: docs/source/nlp/megatron_onnx_export.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+This guide demonstrates the usage of the ONNX export functionality for Megatron models.
+
+Requirements
+-----------------
+Set up the development environment by launching the latest NeMo container
+
+.. code-block:: bash 
+
+    docker run -it nvcr.io/nvidia/nemo:23.04
```
Comment (ericharper) [inline docs/source/nlp/megatron_onnx_export.rst:14]: I don't think this part is needed.

Comment (asfiyab-nvidia) [reply docs/source/nlp/megatron_onnx_export.rst]: updated

**Review by ericharper — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**